server/CVCMIServer.cpp

```diff
@@ -835,6 +835,19 @@ void CVCMIServer::setPlayer(PlayerColor clickedColor)
 	}
 }
 
+void CVCMIServer::setPlayerName(PlayerColor color, std::string name)
+{
+	PlayerSettings & player = si->playerInfos[color];
```

Comment:
```suggestion
	PlayerSettings & player = si->playerInfos.at(color);
```

Also, maybe test if color is valid player (CANNOT_DETERMINE is default value in cpack)